javascript/GridField.js

```diff
@@ -146,6 +152,37 @@
 			}
 		});
 
+		/**
+		 * Don't allow users to submit empty values in grid field auto complete inputs.
+		 */
+		$('.ss-gridfield .add-existing-autocompleter').entwine({
```

Comment:
You have a lot of .find('input') in this class, plus you are having to bind an event to a child element. Why not promote the textbox to the top level.

`$('.ss-gridfield .add-existing-autocompleter input[type="text"]')`

Then you can use `onkeyup` as a top level event. You can still use `onmatch` to do your initial loading, and `$button = this.parents('.add-existing-autocompleter`).find('input[type="submit"]')` to get the button.